src/common/transformations/src/transformations/common_optimizations/nonzero_horizontal_fusion.cpp

```diff
@@ -24,8 +24,9 @@ ov::pass::NonZeroFusion::NonZeroFusion() {
 
         bool status = false;
         auto replace_if_nodes_match = [&](const ov::Input<ov::Node>& in) {
-            auto cur_nonzero = ov::as_type_ptr<ov::opset10::NonZero>(in.get_node()->shared_from_this());
-            if (cur_nonzero && cur_nonzero->get_output_type() == out_prc) {
+            auto in_node = in.get_node()->shared_from_this();
```

Comment:
why do you need to call ->shared_from_this()
I believe you can use `in_node` itself without .get()

Reply:
Yes, you are right. Corrected, thanks